rmf_fleet_adapter/src/rmf_fleet_adapter/events/GoToPlace.hpp

```diff
@@ -103,13 +106,22 @@ class GoToPlace : public rmf_task_sequence::Event
     void kill() final;
 
   private:
+    enum class ReservationState
+    {
+      Pending=0,
+      Requested=1,
+      RecievedResponse=2
+    };
+
+    ReservationState _current_reservation_state = ReservationState::Pending;
+
 
     Active(Description description);
 
     void _schedule_retry();
 
     std::optional<rmf_traffic::agv::Plan::Goal> _choose_goal(
-      bool only_same_map) const;
+      bool only_same_map);
```

Comment:
I haven't looked in depth but it seems this function should still be able to be `const`? It seems to just iterate through possible goals and choose the closest one